pyhf/pdf.py

```diff
@@ -130,6 +130,42 @@ def __init__(self, spec, **config_kwargs):
         utils.validate(self.spec, self.schema)
         # build up our representation of the specification
         self.config = _ModelConfig.from_spec(self.spec,**config_kwargs)
+        self.cube = self._make_cube()
+
+
+    def _make_cube(self):
+        """
+        Dear Lukas, maybe we don't need NaNs.
+
+        self.cube only directly gets added to deltas first, before we do any
+        multiplication with factors. If this is true, then setting any
+        situation where we have NaNs to zero will work because we'll add before
+        doing anything else.
+
+        Similarly, if this is not always true, we can put NaNs back in
+        (uncomment the lines here) and use tensorlib.sum(...,nansum=True) to
+        get the equivalent sum over nans which should provide the same result.
+
```

Comment:
this is probably true, as it corresponds to additional "spurious samples" with no entries in the bins